browser/ui/views/omnibox/brave_search_conversion_promotion_view.cc

```diff
@@ -0,0 +1,589 @@
+// OmniboxResultSelectionIndicator
+
+class BraveOmniboxResultSelectionIndicator : public views::View {
+ public:
+  METADATA_HEADER(BraveOmniboxResultSelectionIndicator);
+
+  static constexpr int kStrokeThickness = 3;
+
+  explicit BraveOmniboxResultSelectionIndicator(
+      BraveSearchConversionPromotionView* parent_view)
+      : parent_view_(parent_view) {
+    SetPreferredSize({kStrokeThickness, 0});
+  }
+
+  // views::View:
+  void OnPaint(gfx::Canvas* canvas) override {
+    SkPath path = GetPath();
+    cc::PaintFlags flags;
+    flags.setAntiAlias(true);
+    flags.setColor(color_);
+    flags.setStyle(cc::PaintFlags::kFill_Style);
+    canvas->DrawPath(path, flags);
+  }
+
+  // views::View:
+  void OnThemeChanged() override {
+    views::View::OnThemeChanged();
+
+    color_ = views::GetCascadingAccentColor(parent_view_);
+  }
+
+ private:
+  SkColor color_;
+
+  // Pointer to the parent view.
+  const raw_ptr<BraveSearchConversionPromotionView> parent_view_;
+
+  // The focus bar is a straight vertical line with half-rounded endcaps. Since
+  // this geometry is nontrivial to represent using primitives, it's instead
+  // represented using a fill path. This matches the style and implementation
+  // used in Tab Groups.
+  SkPath GetPath() const {
+    SkPath path;
+
+    path.moveTo(0, 0);
+    path.arcTo(kStrokeThickness, kStrokeThickness, 0, SkPath::kSmall_ArcSize,
+               SkPathDirection::kCW, kStrokeThickness, kStrokeThickness);
+    path.lineTo(kStrokeThickness, height() - kStrokeThickness);
+    path.arcTo(kStrokeThickness, kStrokeThickness, 0, SkPath::kSmall_ArcSize,
+               SkPathDirection::kCW, 0, height());
+    path.close();
+
+    return path;
+  }
+};
+
+BEGIN_METADATA(BraveOmniboxResultSelectionIndicator, views::View)
+END_METADATA
+
+}  // namespace
+
+BraveSearchConversionPromotionView::BraveSearchConversionPromotionView(
+    BraveOmniboxResultView* result_view)
+    : result_view_(result_view),
+      mouse_enter_exit_handler_(base::BindRepeating(
+          &BraveSearchConversionPromotionView::UpdateHoverState,
+          base::Unretained(this))) {
+  SetLayoutManager(std::make_unique<views::FlexLayout>());
+}
+
+BraveSearchConversionPromotionView::~BraveSearchConversionPromotionView() =
+    default;
+
+void BraveSearchConversionPromotionView::ResetChildrenVisibility() {
+  // Reset all children visibility and configure later based on type.
+  if (button_type_container_)
+    button_type_container_->SetVisible(false);
+
+  if (banner_type_container_)
+    banner_type_container_->SetVisible(false);
+}
+
+void BraveSearchConversionPromotionView::SetTypeAndInput(
+    brave_search_conversion::ConversionType type,
+    const std::u16string& input) {
+  DCHECK_NE(ConversionType::kNone, type);
+
+  type_ = type;
+  input_ = input;
+  selected_ = false;
+
+  ResetChildrenVisibility();
+
+  if (type_ == ConversionType::kButton) {
```

Comment:
Would it be more simple and easy to read to have 2 separate views - one for the "button" type and another for the "banner" type? Either as children for this view or as replacement for this view.

Reply:
Actually, this view has two childrens for each types.
`button_type_container_` and `banner_type_container_` for button and banner type.
Based on type, only one is visibile in this promotion view. Added comment about this - https://github.com/brave/brave-core/pull/12846/commits/68d8c817ae47c4b980b9fd245cca2b9d35f41819